src/ui/GenioWindow.cpp

```diff
@@ -3548,6 +3554,17 @@ GenioWindow::_ShowInTracker(const entry_ref& ref)
 }
 
 
+void
+GenioWindow::_SelectInTracker(const entry_ref& ref)
+{
+	BMessenger tracker("application/x-vnd.Be-TRAK");
+	BMessage selectMessage('Tsel');
+	selectMessage.AddRef("refs", &ref);
+	snooze(300000); // wait 0.3 sec to give Tracker time to populate
```

Comment:
What about if we use a BMessageRunner here? with one shot after 0.3 (or 0.5 at this point) and avoid any locking on GenioWindow.

Reply:
I doubt anyone can actual see Genio freeze for 0.3 seconds. The previous _ShowInTracker() will open and put the folder's Tracker window in focus. Only then occurs the 0.3s wait until the file/folder is selected. By the time the user can react to make the Genio window get the focus back, the file is already selected and the Genio window is reanimated.

Reply:
It's not only a problem of freezing the UI, GenioWindow is also used to dispatch LSP and Path Monitoring messaging and I would like to avoid to create any queue. Anyway is more a best-practice than a real need. Moreover the code is more elegant IMHO:

```
	BMessage selectMessage('Tsel');
	selectMessage.AddRef("refs", &ref);
	BMessageRunner::StartSending(BMessenger("application/x-vnd.Be-TRAK"),
								&selectMessage, 300000, 1);

```

Reply:
Thanks for that hint, I wasn't aware of BMessageRunner::StartSending()!
I updated the PR accordingly.